src/engine/srv.c

```diff
@@ -1322,14 +1320,21 @@ dss_srv_init(void)
 		D_GOTO(failed, rc);
 	xstream_data.xd_init_step = XD_INIT_TLS_INIT;
 
+	if (dss_nvme_conf) {
+		rc = bio_check_md_on_ssd(dss_nvme_conf);
+		if (rc < 0)
+			D_GOTO(failed, rc);
+		if (rc > 0)
+			dss_md_on_ssd_enabled = true;
+	}
 	rc = dss_sys_db_init(&db);
 	if (rc != 0)
 		D_GOTO(failed, rc);
 	xstream_data.xd_init_step = XD_INIT_SYS_DB;
 
 	rc = bio_nvme_init(dss_nvme_conf, dss_numa_node, dss_nvme_mem_size,
 			   dss_nvme_hugepage_size, dss_tgt_nr, db,
-			   dss_nvme_bypass_health_check);
+			   dss_nvme_bypass_health_check, dss_md_on_ssd_enabled);
```

Comment:
It looks to me that if we reorder the sysdb init and nvme init, the code could be much more simpler:
- bio_nvme_init() is called first, and the config parsing for md on SSD could be done in bio_spdk_env_init() along with other parameter parsing. (hotplug filter, accel, rpc server).
- move smd_init() from bio_nvme_init() to dss_sys_db_init(), so that sys db init could be called after bio_nvme_init().
- The global dss_md_on_ssd_enabled could be replaced by bio_nvme_configured() call.
- No need to export new BIO API bio_check_md_on_ssd().

Reply:
Good idea!